src/models/types.ts

```diff
@@ -81,3 +81,15 @@ export interface Authority {
   mint: number;
   melt: number;
 }
+
+/**
+ * A backend-mediated Atomic Swap proposal, containing backend metadata such as id, version and history
+ */
+export interface AtomicSwapProposal {
+  proposalId: string,
+  partialTx: string,
+  signatures: string | null,
```

Comment:
Is there a case in which the data can prescind a signature? Could you give an example?

Reply:
While the proposal is still in the negotiations phase ( inputs and outputs are being added/edited/removed ), there is no signature data on the `AtomicSwapProposal` object.

Only when all participants agree on the proposal contents and the `PartialTx` is considered complete the signatures start to be collected. You can read more about that on our [GitBook section](https://hathor.gitbook.io/hathor/guides/headless-wallet/atomic-swap#sequence-of-steps) about the atomic swap on the headless.